Magitek/Logic/Warrior/SingleTarget.cs

```diff
@@ -157,7 +157,8 @@ public static async Task<bool> Onslaught()
                 return false;
             }
 
-            if (Spells.Onslaught.Charges < WarriorSettings.Instance.SaveOnslaughtCharges + 1)
+            //Why + 1 just keep the number of charges specified - also if you want to keep that number it should be <= not just <
```

Comment:
I think this is because Spells.Onslaught.Charges is a float and it increases fractionally as a charge goes through its cooldown.

With your change if someone has SaveOnslaughtCharges set to 1, as soon as Charges goes to 1.1 then a charge would be used, which would result in the player not having a charge available any more despite a setting that indicated they wanted to save 1 charge.